src/core/lib/resource_quota/memory_quota.cc

```diff
@@ -469,9 +473,52 @@ BasicMemoryQuota::InstantaneousPressureAndMaxRecommendedAllocationSize() const {
   double pressure = (size - free) / size;
   if (pressure < 0.0) pressure = 0.0;
   if (pressure > 1.0) pressure = 1.0;
+  if (kSmoothMemoryPressure) {
+    pressure = pressure_tracker_.AddSampleAndGetEstimate(pressure);
+  }
   return std::make_pair(pressure, quota_size / 16);
 }
 
+//
+// PressureTracker
+//
+
+namespace memory_quota_detail {
+
+double PressureTracker::AddSampleAndGetEstimate(double sample) {
+  double max_so_far = max_this_round_.load(std::memory_order_relaxed);
+  if (sample > max_so_far) {
+    max_this_round_.compare_exchange_weak(max_so_far, sample,
+                                          std::memory_order_relaxed,
+                                          std::memory_order_relaxed);
+  }
+  // If memory pressure is almost done, immediately hit the brakes and report
+  // full memory usage.
+  if (sample >= 0.99) {
+    report_.store(1.0, std::memory_order_relaxed);
+  }
+  update_.Tick([&](Duration dt) {
+    // Reset the round tracker with the new sample.
+    const double current_estimate =
+        max_this_round_.exchange(sample, std::memory_order_relaxed);
+    double report;
+    if (current_estimate > 0.99) {
+      // Under very high memory pressure we... just max things out.
+      report = pid_.Update(1e99, 1.0);
+    } else {
+      report = pid_.Update(current_estimate - 0.8, dt.seconds());
```

Comment:
Should the return value be reported only when current_estimate > 0.8 ? I'm not sure what the return value represents if current_estimate is less than 0.8. The PID controller cannot do anything to increase memory usage right

Reply:
I think this is right... if the PID controller sets the control value (`report` here) to zero, it's going to effectively say 'use as much memory as is possible' which ought to increase memory usage after some delay if it's possible to do so.

Reply:
Yeah, I don't have a good intuition on where the set point should be. Added a config, let's toy with it on some real projects and see where we get to.

Reply:
(also lowered to 60% for the default initially)

Reply:
Thanks. I think memory_quota_test is failing

Reply:
fixed by tweaking pid controller weights (which we'll probably be turning for a bit)